File: src/server/routes/lifecycle.js
// akiratv/src/server/routes/lifecycle.js
// JavaScript port of Python routes/lifecycle.py

const express = require('express');
const router = express.Router();
const { getAkiraTV } = require('../app');

/**
 * Start AkiraTV engine
 */
router.post('/start', async (req, res) => {
    try {
        const akiraTV = getAkiraTV();
        if (!akiraTV) {
            return res.status(500).json({ success: false, error: 'AkiraTV not initialized' });
        }
        
        const result = await akiraTV.start();
        if (result) {
            res.json({ success: true, message: 'AkiraTV started successfully' });
        } else {
            res.status(500).json({ success: false, error: 'Failed to start AkiraTV' });
        }
    } catch (e) {
        res.status(500).json({ success: false, error: e.message });
    }
});

/**
 * Stop AkiraTV engine
 */
router.post('/stop', async (req, res) => {
    try {
        const akiraTV = getAkiraTV();
        if (!akiraTV) {
            return res.status(500).json({ success: false, error: 'AkiraTV not initialized' });
        }
        
        akiraTV.stop();
        res.json({ success: true, message: 'AkiraTV stopped successfully' });
    } catch (e) {
        res.status(500).json({ success: false, error: e.message });
    }
});

/**
 * Restart AkiraTV engine
 */
router.post('/restart', async (req, res) => {
    try {
        const akiraTV = getAkiraTV();
        if (!akiraTV) {
            return res.status(500).json({ success: false, error: 'AkiraTV not initialized' });
        }
        
        // Stop first
        akiraTV.stop();
        
        // Wait a moment then start
        await new Promise(resolve => setTimeout(resolve, 1000));
        
        const result = await akiraTV.start();
        if (result) {
            res.json({ success: true, message: 'AkiraTV restarted successfully' });
        } else {
            res.status(500).json({ success: false, error: 'Failed to restart AkiraTV' });
        }
    } catch (e) {
        res.status(500).json({ success: false, error: e.message });
    }
});

/**
 * Get engine status
 */
router.get('/status', (req, res) => {
    const akiraTV = getAkiraTV();
    if (!akiraTV) {
        return res.json({
            is_running: false,
            uptime: '0s',
            stats: {}
        });
    }
    
    const stats = akiraTV.getStats();
    res.json({
        is_running: akiraTV.running,
        uptime: stats.uptime || '0s',
        stats: stats
    });
});

module.exports = router;

File: akiratv/routes/lifecycle.py
"""
Lifecycle routes for AkiraTV API
Handles engine start, stop, restart, and status endpoints
"""

from fastapi import APIRouter, HTTPException, Depends
from ..models import Response
from ..core_api import get_api

router = APIRouter(prefix="/api", tags=["Lifecycle"])

# Dependency for Core API access
def get_core_api():
    """Get CoreAPI instance"""
    return get_api()

@router.post("/start", response_model=Response)
def start_engine(api = Depends(get_core_api)):
    """Start AkiraTV engine"""
    result = api.start()
    if result["success"]:
        return Response(success=True, message=result["message"])
    else:
        raise HTTPException(status_code=500, detail=result["error"])

@router.post("/stop", response_model=Response)
def stop_engine(api = Depends(get_core_api)):
    """Stop AkiraTV engine"""
    result = api.stop()
    if result["success"]:
        return Response(success=True, message=result["message"])
    else:
        raise HTTPException(status_code=500, detail=result["error"])

@router.post("/restart", response_model=Response)
def restart_engine(api = Depends(get_core_api)):
    """Restart AkiraTV engine"""
    result = api.restart()
    if result["success"]:
        return Response(success=True, message=result["message"])
    else:
        raise HTTPException(status_code=500, detail=result["error"])

@router.get("/status")
def get_status(api = Depends(get_core_api)):
    """Get engine status"""
    return {
        "is_running": api.is_running,
        "uptime": api.uptime,
        "stats": api.stats
    }
